roles/kubernetes-apps/ansible/tasks/main.yml

```diff
@@ -1,7 +1,7 @@
 ---
 - name: Kubernetes Apps | Wait for kube-apiserver
   uri:
-    url: "{{ kube_apiserver_endpoint }}/healthz"
+    url: "{{ external_apiserver_endpoint }}/healthz"
```

Comment:
Why is this necessary? This runs on kube-master[0], which has kube_apiserver_endpoint set.

This adds a strong dependency on kubernetes/client role because the set_fact is created there.

Reply:
Because AFAIK this file will be used while doing the actual kubectl apply`s:
https://github.com/kubernetes-incubator/kubespray/blob/8aafe64397c8d43d44cca49be0b69a461d112af7/roles/kubernetes/client/templates/admin.conf.j2#L8

As you can see this file uses external endpoint instead of the internal one. The waiting however happens on the internal one so it can cause inconsistencies. 

So internal will be set to "localhost..." external maybe to "mydomain.com" but mydomain.com is not reachable yet because we tested localhost instead of mydomain.

Might also be that I`m missing something, in that case please clarify.

Reply:
Any update on this, we would rather not have to maintain a local fork ?